functions/geometry_kube.zsh

```diff
@@ -3,15 +3,28 @@
 (( $+commands[kubectl] )) || return
 
 geometry_kube() {
-  ( ${GEOMETRY_KUBE_PIN:=false} ) || [[ -f ~/.kube/config ]] || return
+  ( ${GEOMETRY_KUBE_PIN:=false} ) || [[ ! -z "${KUBECONFIG}" ]] || return
 
-  GEOMETRY_KUBE=$(ansi ${GEOMETRY_KUBE_COLOR:=blue} ${GEOMETRY_KUBE_SYMBOL:="⎈"})
+  local geometry_kube=$(ansi ${GEOMETRY_KUBE_COLOR:=blue} ${GEOMETRY_KUBE_SYMBOL:="⎈"})
+  local geometry_kube_separator=${GEOMETRY_KUBE_SEPARATOR:="|"}
 
-  [[ $(kubectl version --client --short) =~ 'Client Version: ([0-9a-zA-Z.]+)' ]]
-  local version=$match[1]
+  if ( ${GEOMETRY_KUBE_VERSION:=true} ); then
+    [[ $(kubectl version --client --short) =~ 'Client Version: ([0-9a-zA-Z.]+)' ]]
+    geometry_kube+=($match[1])
+  fi
 
-  local context="$(kubectl config current-context 2> /dev/null)"
-  local namespace="$(kubectl config view -o "jsonpath={.contexts[?(@.name==\"$context\")].context.namespace}" 2> /dev/null)"
+  local geometry_context="$(kubectl config current-context 2> /dev/null)"
+  local geometry_namespace="$(kubectl config view -o "jsonpath={.contexts[?(@.name==\"$context\")].context.namespace}" 2> /dev/null)"
 
-  echo -n "$GEOMETRY_KUBE $version ($context:${namespace:=default})"
+  if [[ ! -z "${GEOMETRY_KUBE_CONTEXT_COLOR}" ]]; then
+    geometry_context="$(ansi ${GEOMETRY_KUBE_CONTEXT_COLOR:=white} ${geometry_context})"
+  fi
+
+  if [[ ! -z "${GEOMETRY_KUBE_NAMESPACE_COLOR}" ]]; then
```

Comment:
does this make `${GEOMETRY_KUBE_NAMESPACE_COLOR:=white}` redundant?

I think this entire test and else branch can be removed and have the same effect?

Reply:
I'd also be open to switching `ansi`'s argument order, and making the second argument optional. This would let us possibly remove all of the 'white' defaults easier.

Reply:
though that kinda messes with the ability to have text with spaces. Which ansi breaks right now too I guess?

Reply:
Okay, white can be changed to `default`